libs/database/src/lib/services/user.service.v2.ts

```diff
@@ -209,4 +211,116 @@ export class UserServiceV2 {
+    const blockUser = await this.repositoryService
+      .findUser({
+        _id: targetCastcleId,
+      })
+      .exec();
+
+    if (!blockUser) throw CastcleException.USER_OR_PAGE_NOT_FOUND;
+
+    await Promise.all([
+      this.repositoryService.updateRelationship(
+        { user: user._id, followedUser: blockUser._id },
+        {
+          $setOnInsert: {
+            user: user._id,
+            followedUser: blockUser._id,
+            visibility: EntityVisibility.Publish,
+            following: false,
+            blocked: false,
+          },
+          $set: { blocking: true },
+        },
+        { upsert: true }
+      ),
+      this.repositoryService.updateRelationship(
+        { followedUser: user._id, user: blockUser._id },
+        {
+          $setOnInsert: {
+            followedUser: user._id,
+            user: blockUser._id,
+            visibility: EntityVisibility.Publish,
+            following: false,
+            blocking: false,
+          },
+          $set: { blocked: true },
+        },
+        { upsert: true }
+      ),
+    ]);
+  }
+
+  async getBlockedLookup(
+    user: User,
+    { hasRelationshipExpansion, maxResults, sinceId, untilId }: PaginationQuery
+  ) {
+    const filterQuery = {
+      sinceId,
+      untilId,
+      userId: [user._id],
+      blocking: true,
+    };
+
+    const relationships = await this.repositoryService
+      .findRelationships(filterQuery)
+      .sort({ followedUser: SortDirection.DESC })
+      .limit(maxResults)
+      .exec();
+
+    const userIds = relationships.map(({ followedUser }) => followedUser);
+
+    return this.getByCriteria(
+      user,
+      { _id: userIds },
+      {},
+      hasRelationshipExpansion
+    );
+  }
+
+  getByCriteria = async (
+    viewer: User,
+    query: FilterQuery<User>,
+    queryOptions?: CastcleQueryOptions,
+    hasRelationshipExpansion = false,
+    userFields?: UserField[]
+  ) => {
+    const { items: targetUsers, meta } = await this.getAllByCriteria(
+      query,
+      queryOptions
+    );
+
+    const users = await this.convertUsersToUserResponsesV2(
+      viewer,
+      targetUsers,
+      hasRelationshipExpansion,
+      userFields
+    );
+    return { users, meta };
+  };
+
+  getAllByCriteria = async (
+    query: FilterQuery<User>,
+    queryOptions?: CastcleQueryOptions
+  ) => {
+    const filterQuery = { ...query };
+    const total = await this.repositoryService.findUserCount(filterQuery);
+    let usersQuery = this.repositoryService.findUsers(filterQuery as any);
+    if (queryOptions?.limit) usersQuery = usersQuery.limit(queryOptions.limit);
+    if (queryOptions?.page) usersQuery = usersQuery.skip(queryOptions.page - 1);
+    if (queryOptions?.sortBy) {
+      const sortDirection = queryOptions.sortBy.type === 'desc' ? '-' : '';
+      const sortOrder = `${sortDirection}${queryOptions.sortBy.field}`;
+      usersQuery = usersQuery.sort(sortOrder);
+    }
```

Comment:
```suggestion
    if (queryOptions?.limit) usersQuery = usersQuery.limit(queryOptions.limit);
    if (queryOptions?.page) usersQuery = usersQuery.skip(queryOptions.page - 1);
    if (queryOptions?.sortBy) {
      const sortDirection = queryOptions.sortBy.type === 'desc' ? '-' : '';
      const sortOrder = `${sortDirection}${queryOptions.sortBy.field}`;
      usersQuery = usersQuery.sort(sortOrder);
    }
```
ตรงนี้ ไม่แน่ใจว่ามีค่า default อยู่แล้วหรือป่าวครับ ถ้ามีแล้วไม่ต้อง check ก็ได้ครับ